appcelerator/titanium_mobile
# fix(android): Fixes changing background colors for Tabs after they are drawn.

**JIRA:** https://jira.appcelerator.org/browse/TIMOB-26859

**Description:**
Fixes changing TabGroup and Tab background colors for different states after the creation of the
components. 

Note: No unit tests. Also I see that in the code sample in the ticket the property `backgroundSelectedColor` (https://docs.appcelerator.com/platform/latest/#!/api/Titanium.UI.Tab-property-backgroundSelectedColor) is being used. When I was refactoring the TabGroup I went for the usage of `backgroundFocusedColor` (https://docs.appcelerator.com/platform/latest/#!/api/Titanium.UI.Tab-property-backgroundFocusedColor) as it explicitly states that is used for changing the color of the Tab according to the state. Do you think we can adjust the docs to just list the latter and add to the description that it acts as `backgroundSelectedColor`?  

**Test Case 1:**
_This tests empty tab group, top/bottom tab color handling, and setting colors after creation but before open._
1. Build and run the below code on Android.
2. Verify that an empty window with no tabs appears. (Must not crash.)
3. Uncomment the `// tabs: ...` line in the code below.
4. Rebuild and run on Android.
5. Verify that left-most tab (the selected tab) is yellow and all others tabs are green.
6. Tap on center tab and verify that it turns yellow and left tab turns green.
7. Uncomment the `// style: ...` line in the code below.
8. Rebuild and run on Android.
9. Verify that the tabs are now docked to the bottom of the window instead of the top.
10. Verify that left-most tab (the selected tab) is yellow and all others tabs are green.
11. Tap on center tab and verify that it turns yellow and left tab turns green.

```javascript
function createTab(title) {
	var window = Ti.UI.createWindow({ title: title });
	window.add(Ti.UI.createLabel({ text: title + " View" }));
	return Ti.UI.createTab({ title: title, window: window });
}
var tabGroup = Ti.UI.createTabGroup({
//	tabs: [createTab("Tab 1"), createTab("Tab 2"), createTab("Tab 3")],
//	style: Ti.UI.Android.TABS_STYLE_BOTTOM_NAVIGATION,
	tabsBackgroundColor: "red",
	tabsBackgroundSelectedColor: "orange",
});
tabGroup.tabsBackgroundColor = "green";
tabGroup.tabsBackgroundSelectedColor = "yellow";
tabGroup.open();
```

**Test Case 2:**
_This tests changing tab colors after opening tab group._
```js
// create tab group
var tabGroup = Titanium.UI.createTabGroup({
	// Test tabsBackgroundColor and tabsBackgroundSelectedColor here.
	tabsBackgroundSelectedColor:'red',
	// Comment/uncomment the bottom line to check both styles.
	// style: Ti.UI.Android.TABS_STYLE_BOTTOM_NAVIGATION,
	tabsBackgroundColor:'blue',
});

var win1 = Titanium.UI.createWindow({
	title:'Tab 1',
	backgroundColor:'#fff'
});

var tab1 = Titanium.UI.createTab({
	backgroundColor: 'lime',
	backgroundFocusedColor: 'red',
	title:'Tab 1',
	activeTitleColor: 'black',
	window:win1
});

var label1 = Titanium.UI.createLabel({
	color:'#999',
	text:'I am Window 1',
	font:{fontSize:20,fontFamily:'Helvetica Neue'},
	textAlign:'center',
	width:'auto'
});

win1.add(label1);

// create controls tab and root window
var win2 = Titanium.UI.createWindow({
	title:'Tab 2',
	backgroundColor:'#fff'
});

var tab2 = Titanium.UI.createTab({
	title:'Tab 2',
	titleColor: 'black',
	window:win2
});

var label2 = Titanium.UI.createLabel({
	color:'#999',
	text:'I am Window 2',
	font:{fontSize:20,fontFamily:'Helvetica Neue'},
	textAlign:'center',
	width:'auto'
});

win2.add(label2);

var win3 = Titanium.UI.createWindow({
	title:'Tab 3',
	backgroundColor:'#fff'
});

var tab3 = Titanium.UI.createTab({
	title:'Tab 3',
	titleColor: 'black',
	window:win3
});

var button3 = Titanium.UI.createButton({
	title: "change background color"
});

button3.addEventListener('click', function(e){
	// Test tabsBackgroundColor and tabsBackgroundSelectedColor here.
	tabGroup.tabsBackgroundColor='orange';
	tabGroup.tabsBackgroundSelectedColor = 'gray';
	tab2.backgroundColor = 'cyan',
	tab2.backgroundFocusedColor = 'magenta'
});

win2.add(button3);

// add tabs
tabGroup.addTab(tab1);
tabGroup.addTab(tab2);
tabGroup.addTab(tab3);
// open tab group
tabGroup.open();

```
